datahub-web-react/build.gradle

```diff
@@ -67,36 +72,44 @@ task yarnBuild(type: YarnTask, dependsOn: [yarnInstall, yarnTest, yarnLint]) {
   args = ['run', 'build']
 }
 
-clean {
+task yarnQuickBuild(type: YarnTask, dependsOn: [yarnInstall, yarnGenerate]) {
+  args = ['run', 'build']
+}
+
+task cleanExtraDirs {
   delete 'node_modules'
   delete 'dist'
   delete 'tmp'
-  delete 'build'
   delete 'just'
   delete 'src/types.generated.ts'
+  delete fileTree('../datahub-frontend/public')
   delete fileTree(dir: 'src/graphql', include: '*.generated.ts')
 }
+clean.finalizedBy(cleanExtraDirs)
 
 configurations {
   assets
 }
 
 distZip {
-  dependsOn yarnBuild
+  dependsOn yarnQuickBuild
   baseName 'datahub-web-react'
   from 'dist'
 }
 
-task copyAssets {
-  dependsOn distZip
-  copy {
-    from 'dist'
-    into '../datahub-frontend/public/'
+task copyAssets(dependsOn: distZip) {
```

Comment:
was the copy task in datahub-frontend buid.gradle just not doing anything?

Reply:
This reduces the cross subproject task dependency (gradle documentation has this called out as bad practice). This dependency is now expressed as a more natural implementation dependency (since the jar here has assets it needs) and the copy of these static files is merely a side-effect of that dependency. The static files are only used for locally running the frontend and unit tests. At runtime in the container the assets are loaded from the classpath from the react jar by the play framework.